pkg/api/v1/control-plane/cache/prioritylist.go

```diff
@@ -0,0 +1,165 @@
+package cache
+
+import (
+	"fmt"
+	"sort"
+)
+
+// The default Priority is the last index. The last index is not required in the prioirty list, thus any elements added
+// to the struct will be added to the default priority (last).
+const defaultPrioirty = -1
```

Comment:
defaultpriority could just be 0 and not a different const?

 see https://go.dev/play/p/k5akHGKRAwf

Reply:
0 is an index in the priority map

Reply:
```
	if priority == defaultPrioirty {
		// add to the last index of the watches
		priority = p.numberOfPriorities
	}
```